Code/Editor/CustomTemplatePaths.cpp

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include "EditorCommon.h"
+#include "CustomTemplatePaths.h"
+#include "ui_CustomTemplatePaths.h"
+
+#include <QString>
+#include <QListWidgetItem>
+
+CustomTemplatePaths::CustomTemplatePaths(QWidget* parent) : QDialog(parent), ui(new Ui::CustomTemplatePaths)
+{
+    ui->setupUi(this);
+
+    setStyleSheet("QListWidget::item {height: 16px; padding-left: 0px; padding-right: 0px; background-color: transparent;}");
+
+    QSettings settings(QSettings::IniFormat, QSettings::UserScope, "O3DE");
+    settings.beginGroup("CustomTemplatePaths");
+    lastSelectedFolderPath = settings.value("lastSelectedFolderPath").toString();
+    settings.endGroup();
+
+    if(lastSelectedFolderPath.isEmpty())
+    {
+        lastSelectedFolderPath = (AZ::Utils::GetProjectPath().c_str());
+    }
+
+    QObject::connect(ui->pushButtonAdd, &QPushButton::clicked, this, [this]()
+    {
+        QString folderPath = QFileDialog::getExistingDirectory(this, tr("Select Folder"), lastSelectedFolderPath);
+        if (!folderPath.isEmpty())
+        {
+            QFileInfo info(folderPath);
+            auto* item = new QListWidgetItem(info.baseName());
+            item->setIcon(QIcon(":/NewLevel/res/folder.png"));
+            item->setData(Qt::UserRole, folderPath);
+            item->setToolTip(folderPath);
+            ui->listWidgetTemplatePaths->addItem(item);
+
+            lastSelectedFolderPath = folderPath;
+
+            SaveSettings();
+        }
+    });
+
+    QObject::connect(ui->pushButtonRemove, &QPushButton::clicked, this, [this]()
+    {
+        if (const QListWidgetItem* selectedItem = ui->listWidgetTemplatePaths->currentItem())
+        {
+            int index = ui->listWidgetTemplatePaths->row(selectedItem);
+            ui->listWidgetTemplatePaths->takeItem(index);
+
+            SaveSettings();
+        }
+    });
+
+    QObject::connect(ui->pushButtonOk, &QPushButton::clicked, this, &QDialog::accept);
+
+    LoadSettings();
+}
+
+void CustomTemplatePaths::LoadSettings() const
+{
+    QSettings settings(QSettings::IniFormat, QSettings::UserScope, "O3DE");
+    settings.beginGroup("CustomTemplatePaths");
+    const QStringList folderPaths = settings.value("folderPaths").toStringList();
+    settings.endGroup();
+
+    for (auto& Item: folderPaths)
+    {
+        QFileInfo info(Item);
+        auto* item = new QListWidgetItem(info.baseName());
+        item->setIcon(QIcon(":/NewLevel/res/folder.png"));
+        item->setData(Qt::UserRole, Item);
+        item->setToolTip(Item);
+        ui->listWidgetTemplatePaths->addItem(item);
+    }
```

Comment:
Not yet so familiar to save to the settings register. Any file I can have a look into?

Reply:
@lemonade-dm any advice on how this could be handled, especially if we wanted to allow Gems and Projects to plug in new templates?

Reply:
ok. i used AZ::SettingsRegistry to get the path from /O3DE/Preferences/Prefabs/CustomTemplatePath. if the user specfies this folder like this:  "CustomTemplatePath": "@projectroot@/Prefabs/SomeCustomPrefabs" it will be also added to the lists inside the dialog.